From 0d3438d8febb30b1c95c883395e8e4244d7142af Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:13 -0500
Subject: [drm] radeon/dpm: disable ulv support on SI

Message-id: <1418852380-13061-63-git-send-email-rclark@redhat.com>
Patchwork-id: 102205
O-Subject: [RHEL7 drm 62/89] drm/radeon/dpm: disable ulv support on SI
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.18

commit 6fa455935ab956248b165f150ec6ae9106210077

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Mon Oct 13 12:44:49 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Mon Dec 15 14:06:13 2014 -0500

    drm/radeon/dpm: disable ulv support on SI

    Causes problems on some boards.

    bug:
    https://bugs.freedesktop.org/show_bug.cgi?id=82889

    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Cc: stable@vger.kernel.org

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/si_dpm.c b/drivers/gpu/drm/radeon/si_dpm.c
index 5891886..78dfbe3 100644
--- a/drivers/gpu/drm/radeon/si_dpm.c
+++ b/drivers/gpu/drm/radeon/si_dpm.c
@@ -6207,7 +6207,7 @@ static void si_parse_pplib_clock_info(struct radeon_device *rdev,
  if ((rps->class2 & ATOM_PPLIB_CLASSIFICATION2_ULV) &&
      index == 0) {
   /* XXX disable for A0 tahiti */
-  si_pi->ulv.supported = true;
+  si_pi->ulv.supported = false;
   si_pi->ulv.pl = *pl;
   si_pi->ulv.one_pcie_lane_in_ulv = false;
   si_pi->ulv.volt_change_delay = SISLANDS_ULVVOLTAGECHANGEDELAY_DFLT;
-- 
1.7.1